hera_mc/cm_hookup.py

```diff
@@ -188,28 +191,54 @@ def __add_hookup_timing(self, hookup_dict):
+        """
+        The columns in the hookup_dict contain parts in the hookup chain and the column headers are 
+        the part types contained in that column.  This returns the headers for the retrieved hookup.
+        This gets the full set of headers for a future show_hookup that doesn't require the same
+        hookup starting point.  Returns a single column for now (the one/long_column stuff).
+        The method searches all of the hookup chains to find the longest one, which occurs at 
+        """
+        lc = cm_utils.utility_method()  # This tracks the part and pol keys for the longest hookup
+        lc.v(part=huh.keys()[0], pol=huh[huh.keys()[0]].keys()[0], hlen=0)
+
+        hu_col = {}
+        return_one_column_for_now = True
+        if return_one_column_for_now:  # This is to just not find part_dossiers on everything for now
+            for hk, hu in huh.iteritems():
+                for pk, pv in hu.iteritems():
+                    if len(pv) > lc.hlen:
+                        lc.v(part=hk, pol=pk, hlen=pv)
+            huh = {hk: {pk: huh[lc.part][lc.pol]}}
+
+        lc.v(hlen=0)
+        for hk, hu in huh.iteritems():
+            hu_col[hk] = {}
+            for pk, pv in hu.iteritems():
+                hu_col[hk][pk] = []
+                for h in pv:
+                    get_part_type = self.handling.get_part_dossier(hpn=h.upstream_part,
+                                                                   rev=h.up_part_rev,
+                                                                   at_date=self.at_date,
+                                                                   exact_match=True)
+                    pr_key = cm_utils._make_part_key(h.upstream_part, h.up_part_rev)
+                    hu_col[hk][pk].append(get_part_type[pr_key]['part'].hptype)
+                h = pv[-1]
+                get_part_type = self.handling.get_part_dossier(hpn=h.downstream_part,
+                                                               rev=h.down_part_rev,
+                                                               at_date=self.at_date,
+                                                               exact_match=True)
+                pr_key = cm_utils._make_part_key(h.downstream_part, h.down_part_rev)
+                hu_col[hk][pk].append(get_part_type[pr_key]['part'].hptype)
+                hu_col[hk][pk].append('Start')
+                hu_col[hk][pk].append('Stop')
+                # Next two lines for "one column" solution
+                if len(hu_col[hk][pk]) >= lc.hlen:
+                    lc.v(part=hk, pol=pk, hlen=len(hu_col[hk][pk]))
+        return hu_col[lc.part][lc.pol]
 
     def __hookup_add_correlator_levels(self, hookup_dict, testing):
-        warnings.warn("Warning:  correlator levels don't work with new pol hookup scheme yet (CM_HOOKUP[212]).")
+        warnings.warn("Warning:  correlator levels don't work with new pol hookup scheme yet (CM_HOOKUP[210]).")
```

Comment:
View the line number as the initial serial number, which need not change.  Having that tells you which file it is in, then you can unambiguously search for the cryptic string.